### PR TITLE
Replace forEach - push with map.

### PR DESCRIPTION
**Pull request by MifRea1**
and remove "if (true) return true; else return false;"

### CODE REVIEW

**Comment by fchasen** [comment]
Thanks for these! I'll go through and leave some comments on the PR.

I'm a bit hesitant to simplify some of these to one liners are the expense of readability but it is great to have them cleaned up.

File: src/epubcfi.js [hunk 1]
```diff
@@ -152,15 +141,10 @@ EPUBJS.EpubCFI.prototype.parse = function(cfiStr) {
...
-  cfi.steps = [];
-
-  path.forEach(function(part){
-    var step;
-
-    if(part) {
-      step = parseStep(part);
-      cfi.steps.push(step);
-    }
+  cfi.steps = path.filter(function(part) {
```
Comment (fchasen) [inline src/epubcfi.js]: Does this make it loop through all the parts twice? I'm all for replacing the `forEach`s with filter / map but not sure this instance makes sense.

Comment (MifRea1) [inline src/epubcfi.js]: Yes, it does. I've reverted this commit.

**Comment by fchasen** [comment]
Great thank you, I'll review and merge this week.

**Pull request merged**